mate-panel/menu.c

```diff
@@ -265,28 +265,22 @@ static void
 add_drawers_from_alias (MateMenuTreeAlias *alias,
 			const char     *toplevel_id)
 {
-	MateMenuTreeItem *aliased_item;
-
-	aliased_item = matemenu_tree_alias_get_item (alias);
-
-	switch (matemenu_tree_item_get_type (aliased_item)) {
+	switch (matemenu_tree_alias_get_aliased_item_type (alias)) {
 	case MATEMENU_TREE_ITEM_DIRECTORY:
-		add_drawers_from_dir (MATEMENU_TREE_DIRECTORY (aliased_item),
+		add_drawers_from_dir (matemenu_tree_alias_get_directory (alias),
 				      G_MAXINT/2,
 				      toplevel_id);
 		break;
 
 	case MATEMENU_TREE_ITEM_ENTRY:
 		panel_launcher_create_with_id (toplevel_id,
 					       G_MAXINT/2,
-					       matemenu_tree_entry_get_desktop_file_path (MATEMENU_TREE_ENTRY (aliased_item)));
+					       matemenu_tree_entry_get_desktop_file_path (matemenu_tree_alias_get_aliased_entry (alias)));
```

Comment:
I guess the same applies to `matemenu_tree_alias_get_aliased_entry`. This one is even marked as "transfer full".

Reply:
Right... it's used in another place where the entry is added to the list of ref'ed items.